tests/roompanel/TestRoomMob.cpp

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (C) 2023 The MMapper Authors
+
+#include "TestRoomMob.h"
+
+#include <QtTest>
+
+#include "roompanel/RoomMob.h"
+
+TestRoomMob::TestRoomMob() = default;
+
+TestRoomMob::~TestRoomMob() = default;
+
+void TestRoomMob::testInitialization()
+{
+    RoomMobData mobData;
+    QCOMPARE(mobData.getId(), RoomMobData::NOID);
+
+    // Verify that all fields are initialized to empty QVariant
+    for (uint8_t i = 0; i < NUM_MOB_FIELDS; ++i) {
+        QVERIFY(mobData.getField(static_cast<MobFieldEnum>(i)).isNull());
+    }
+}
+
+void TestRoomMob::testSetGetId()
+{
+    RoomMobData mobData;
+    RoomMobData::Id testId = 123;
+    mobData.setId(testId);
+    QCOMPARE(mobData.getId(), testId);
+}
+
+void TestRoomMob::testSetGetField()
+{
+    RoomMobData mobData;
+    QVariant testValue = QVariant::fromValue(QString("TestValue"));
+    mobData.setField(MobFieldEnum::NAME, testValue);
+    QCOMPARE(mobData.getField(MobFieldEnum::NAME), testValue);
+}
+
+void TestRoomMob::testAllocAndUpdate()
+{
+    auto roomMob = RoomMob::alloc();
+    QVERIFY(roomMob != nullptr);
+
+    RoomMobUpdate update;
+    update.setId(roomMob->getId());
+    QVariant newValue = QVariant::fromValue(QString("UpdatedValue"));
```

Comment:
Maybe set the value to "SomeMobName" instead of "UpdatedValue"

Reply:
Done